### PR TITLE
Hides sites with 0% attention from the A-C table

### PR DESCRIPTION
**Pull request by ryanml**
Fixes: https://github.com/brave/brave-browser/issues/2365
Native-ledger: https://github.com/brave-intl/bat-native-ledger/pull/199

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
1. Build from this branch and enable rewards
2. Acquire 0% attention publishers in the AC table (knowingly from an old profile is fine)
    * `publisher_info_db` can be edited directly to add some
3. Ensure that these 0% publishers do not show in the AC table

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by ryanml** [comment]
@jasonrsadler - I updated the bat-native ledger PR to add a `-1` ALL type for the default constructor, as I think there was a case where we needed, also, I fixed a bad entry in `DEPS`. This should be good to retest now. Thanks

**Comment by ryanml** [comment]
@jasonrsadler @NejcZdovc rebased with master - native-ledger updates are now included in this PR

File: vendor/bat-native-ledger/include/bat/ledger/publisher_info.h [hunk 1]
```diff
@@ -52,6 +52,11 @@ LEDGER_EXPORT enum PUBLISHER_EXCLUDE_FILTER {
   FILTER_ALL_EXCEPT_EXCLUDED = 3
 };
 
+LEDGER_EXPORT enum PUBLISHER_PERCENT_FILTER {
```
Comment (bridiver) [inline vendor/bat-native-ledger/include/bat/ledger/publisher_info.h]: why do we need an enum for this?

Comment (bridiver) [reply vendor/bat-native-ledger/include/bat/ledger/publisher_info.h]: seems like we should just default to 100% and the filter should `<=` in the DB

File: components/brave_rewards/browser/publisher_info_database.cc [hunk 2]
```diff
@@ -483,6 +483,10 @@ std::string PublisherInfoDatabase::BuildClauses(int start,
     ledger::PUBLISHER_EXCLUDE_FILTER::FILTER_ALL_EXCEPT_EXCLUDED)
     clauses += " AND pi.excluded != ?";
 
+  if (filter.percent == ledger::PUBLISHER_PERCENT_FILTER::FILTER_ZERO) {
+    clauses += " AND ai.percent != ?";
```
Comment (bridiver) [inline components/brave_rewards/browser/publisher_info_database.cc]: see below - seems like <= would make more sense and just use pass an int (or double or whatever it is) defaulting to 100%

File: vendor/bat-native-ledger/include/bat/ledger/publisher_info.h (hunk 1 above)
Comment (ryanml) [reply vendor/bat-native-ledger/include/bat/ledger/publisher_info.h]: we use enums for both excluded and month in the filter so it seemed to make sense to use one here.

**Comment by ryanml** [comment]
master: `45fd25a9fe8eab2f2a0e8c5bd46b499eca0e810b`

**Pull request merged**